src/main/java/org/example/servletstexts/TextsServlet.java

```diff
@@ -3,89 +3,72 @@
+                    getTextById(pathParts[1], resp);
                 } else {
-                    resp.setStatus(HttpServletResponse.SC_BAD_REQUEST);
-                    respWriter.write("Invalid request path. You entered an extra '/'");
+                    throw new BadRequestException(INVALID_REQUEST_PATH + " You entered an extra '/'");
                 }
             }
         } catch (IOException e1) {
-            resp.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
-            resp.getWriter().write("Internal server error");
+            throw new InternalServerErrorException(INTERNAL_SERVER_ERROR);
         }
     }
 
     @Override
-    protected void doDelete(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+    protected void handleDelete(HttpServletRequest req, HttpServletResponse resp) {
         try (PrintWriter respWriter = resp.getWriter()) {
             String pathInfo = req.getPathInfo();
             if (pathInfo.equals("/deleteAll")) {
                 texts.clear();
+                currentId = 0;
                 respWriter.write("All text has been deleted");
             } else {
                 String[] pathParts = pathInfo.split("/");
-                if (pathParts.length == 2) {
-                    try {
-                        int id = Integer.parseInt(pathParts[1]);
-                        if (texts.remove(id) != null) {
-                            respWriter.write("Text with id " + id + " has been deleted");
-                        } else {
-                            resp.setStatus(HttpServletResponse.SC_BAD_REQUEST);
-                            respWriter.write("Text with id " + id + " not found");
-                        }
-                    } catch (NumberFormatException e) {
-                        resp.setStatus(HttpServletResponse.SC_BAD_REQUEST);
-                        respWriter.write("Invalid format");
-                    }
+                if (pathParts.length == 2 && !pathParts[1].isEmpty()) {
+                    deleteTextById(pathParts[1], resp);
                 } else {
-                    resp.setStatus(HttpServletResponse.SC_BAD_REQUEST);
-                    respWriter.write("Invalid request path. You entered an extra '/'");
+                    throw new BadRequestException(INVALID_REQUEST_PATH + " You entered an extra '/'");
                 }
             }
         } catch (IOException e) {
-            resp.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
-            resp.getWriter().write("Internal server error");
+            throw new InternalServerErrorException(INTERNAL_SERVER_ERROR);
         }
     }
 
     @Override
-    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+    protected void handlePost(HttpServletRequest req, HttpServletResponse resp) {
         resp.setContentType("application/json");
 
         try (PrintWriter respWriter = resp.getWriter()) {
```

Comment:
can it be that you close Writer and try to use this Writer again (for example, in order to process BadRequestException) ?